aws/data_source_aws_dx_private_virtual_interface_test.go

```diff
@@ -0,0 +1,72 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataSourceAwsDxPrivateVirtualInterface_basic(t *testing.T) {
+	key := "DX_CONNECTION_ID"
+	connectionID := os.Getenv(key)
+	if connectionID == "" {
+		t.Skipf("Environment variable %s is not set", key)
+	}
+	vifName := fmt.Sprintf("terraform-testacc-dxvif-%s", acctest.RandString(5))
+	bgpAsn := randIntRange(64512, 65534)
+	vlan := randIntRange(2049, 4094)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDxPrivateVirtualInterfaceConfigBasic(connectionID, vifName, bgpAsn, vlan),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
We can use `resource.TestCheckResourceAttrPair()` for all other the attributes in the schema to ensure they are set in the Terraform state correctly and match the resource. e.g.

```go
resource.TestCheckResourceAttrPair("data.aws_dx_private_virtual_interface.foo", "connection_id", "aws_dx_private_virtual_interface.foo", "connection_id"),
```

Reply:
I think I covered all the attributes with latest round of commits.